pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -107,6 +107,15 @@ func (r *RestartReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
+	// If create/revive db failed, we skip restarting the cluster for redoing create/revive db in vclusterops
```

Comment:
This check should be done before we call `Collect`. Can we move it up?

Reply:
Yes, will move it up